node/forest_libp2p/src/behaviour.rs

```diff
@@ -82,6 +92,29 @@ impl NetworkBehaviourEventProcess<KademliaEvent> for ForestBehaviour {
     }
 }
 
+impl NetworkBehaviourEventProcess<BitswapEvent> for ForestBehaviour {
+    fn inject_event(&mut self, event: BitswapEvent) {
+        match event {
+            BitswapEvent::ReceivedBlock(peer_id, cid, data) => {
+                let cid = cid.to_string();
+                let cid: Cid = Cid::from_str(&cid).unwrap();
```

Comment:
bytes conversion is probably sufficient. The conversion goes `Cid -> bytes -> string -> bytes -> Cid` so can just cut out turning the bytes to string (which just does with multibase)

Probably safe to handle error also (probably just warning about the error and returning) because there could be cases with inconsistencies. iirc they dropped support for v0 Cids which could be one potential dos vector for someone to crash this

Reply:
Hm i agree, but then we can log debug messages for... well... debugging haha

Reply:
> Hm i agree, but then we can log debug messages for... well... debugging haha

Idk what this means, but these still aren't resolved